controllers/topics/index.js

```diff
@@ -292,6 +294,29 @@ const followTopicV2 = async (req, res) => {
   }
 };
 
+const _topicAutoMessage = async (user_id, topic_id, detailTokenUser) => {
+  try {
+    const communityMessageFormat = await CommunityMessageFormat.findOne({
+      where: {topic_id, status: '1'}
+    });
+
+    if (communityMessageFormat && !detailTokenUser.is_anonymous) {
+      await UserTopic.update(
+        {notify_user: true, is_anonymous: detailTokenUser.is_anonymous},
+        {where: {user_id, topic_id}}
+      );
+    } else {
+      await UserTopic.update(
+        {notify_user: false, is_anonymous: detailTokenUser.is_anonymous},
+        {where: {user_id, topic_id}}
+      );
+    }
+  } catch (error) {
+    console.log(error);
+    throw error;
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new function `_topicAutoMessage` has been implemented. However, there are a few issues that need to be addressed:

1. The error handling could be improved. Currently, it just logs the error and rethrows it. It would be better to provide more context about where the error occurred.

2. The use of magic string '1' for status in line 300 should be avoided. Consider using a constant or enum for better readability and maintainability.

3. The update operations on lines 304-307 and 309-312 seem to be almost identical except for the `notify_user` value. This can be simplified by determining the `notify_user` value first and then performing the update operation once.

Here's how you might improve this:

```diff
- const _topicAutoMessage = async (user_id, topic_id, detailTokenUser) => {
-   try {
-     const communityMessageFormat = await CommunityMessageFormat.findOne({
-       where: {topic_id, status: '1'}
-     });
- 
-     if (communityMessageFormat && !detailTokenUser.is_anonymous) {
-       await UserTopic.update(
-         {notify_user: true, is_anonymous: detailTokenUser.is_anonymous},
-         {where: {user_id, topic_id}}
-       );
-     } else {
-       await UserTopic.update(
-         {notify_user: false, is_anonymous: detailTokenUser.is_anonymous},
-         {where: {user_id, topic_id}}
-       );
-     }
-   } catch (error) {
-     console.log(error);
-     throw error;
-   }
- };

+ const STATUS_ACTIVE = '1';
+ 
+ const _topicAutoMessage = async (user_id, topic_id, detailTokenUser) => {
+   try {
+     const communityMessageFormat = await CommunityMessageFormat.findOne({
+       where: {topic_id, status: STATUS_ACTIVE}
+     });
+ 
+     const notify_user = !!(communityMessageFormat && !detailTokenUser.is_anonymous);
+ 
+     await UserTopic.update(
+       {notify_user, is_anonymous: detailTokenUser.is_anonymous},
+       {where: {user_id, topic_id}}
+     );
+   } catch (error) {
+     console.error(`Error in _topicAutoMessage for user ${user_id} and topic ${topic_id}: `, error);
+     throw error;
+   }
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->